server/channels/app/scheduled_post.go

```diff
@@ -41,3 +41,27 @@ func (a *App) SaveScheduledPost(rctx request.CTX, scheduledPost *model.Scheduled
 
 	return savedScheduledPost, nil
 }
+
+func (a *App) GetUserTeamScheduledPosts(rctx request.CTX, userId, teamId string) ([]*model.ScheduledPost, *model.AppError) {
+	if teamId != "" {
+		hasPermissionToTeam := a.HasPermissionToTeam(rctx, userId, teamId, model.PermissionViewTeam)
+		if !hasPermissionToTeam {
+			return nil, model.NewAppError("App.GetUserTeamScheduledPosts", "app.get_user_team_scheduled_posts.team_permission_error", map[string]any{"user_id": userId, "team_id": teamId}, "", http.StatusForbidden)
+		}
+	}
+
+	scheduledPosts, err := a.Srv().Store().ScheduledPost().GetScheduledPostsForUser(userId, teamId)
+	if err != nil {
+		return nil, model.NewAppError("App.GetUserTeamScheduledPosts", "app.get_user_team_scheduled_posts.error", map[string]any{"user_id": userId, "team_id": teamId}, "", http.StatusInternalServerError)
+	}
+
+	if scheduledPosts == nil {
```

Comment:
yes, I'm sure. Store layer is currently inconsistent but the stance at one point was to return nil instead of empty array is no data exists, so that idea is being continued. This is the same as other store layer functions, though there may be some inconsistencies.

Reply:
Fun. That pattern is like how adding `omitempty` all over the struct tags for various json fields really causes problems for anyone consuming our APIs (including us!)